File: courseify-backend/src/main/kotlin/sorsix/project/courseify/domain/Lesson.kt
package sorsix.project.courseify.domain

import jakarta.annotation.Nullable
import jakarta.persistence.*

@Entity
@Table(name = "lesson")
data class Lesson(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long,

    val title: String,

    val description: String,

    @Column(name = "video_title")
    val videoTitle: String,

    @Column(name = "video_url")
    val videoUrl: String,

    @Column(name = "file_title")
    val fileTitle: String,

    @Column(name="file_url")
    val fileUrl: String,

    @ManyToOne
    @JoinColumn(name = "course_id")
    val course: Course,

    @OneToOne
    @Nullable
    @JoinColumn(name = "quiz_id")
    val quiz: Quiz?
) {
}

File: courseify-frontend/src/app/interfaces/lesson.ts
import { Course } from './Course';
import { Quiz } from './Quiz';

export interface Lesson {
  id: number;
  title: string;
  description: string;
  videoTitle: string;
  videoUrl: string;
  fileTitle: string;
  fileUrl: string;
  course: Course;
  quiz: Quiz;
}
